doc: Add documentation note for when LXD 5.0 is already installed (#437)
canonical/microcloud: doc/how-to/install.md

If you don't want to use MicroCloud's full functionality, you can install only some of the snaps.
However, this is not recommended.
```

```{note}

> Depending on you machine OS and state, the LXD snap might already be installed (e.g, you have a version of Ubuntu on the machine having LXD 5.0 already installed).

Reviewer: ```suggestion
Depending on your machines OS and state, the LXD snap might already be installed (e.g you have a version of Ubuntu on the machine with LXD 5.0 already installed).
```

Author: fixed

> In this case instead of doing `sudo snap install lxd --channel=5.21/stable --cohort="+"`, just do `sudo snap remove lxd --purge && sudo snap install lxd --channel=5.21/stable --cohort="+"`

Reviewer: You don't have to uninstall the snap as you can just refresh it:

```suggestion
In this case instead of running `sudo snap install lxd --channel=5.21/stable --cohort="+"`, refresh the snap with `sudo snap refresh lxd --channel=5.21/stable --cohort="+"`
```

Reviewer 2: This does not work. I specifically mentioned issues if refreshing (my second bullet in original post). Something is left behind during the upgrade process. This may be a separate bug, but for now it is what happens.

Reviewer 2: Somehow you get "Invalid project configuration key "features.networks.zones"" if you refresh, but you don't get this error if you purge and install again.

Reviewer 3: I remember this issue. Looking at LXD, `features.networks.zones` does have support in [`5.21`](https://github.com/canonical/lxd/blob/a921fc37579ab2edc2d103f77416bfeb9aba552e/lxd/api_project.go#L1081), but is not supported in [`5.0`](https://github.com/canonical/lxd/blob/04d273193898512178bd46072a0948f7a60a0aa1/lxd/api_project.go#L1078), which suggest the rejecting system was still on `5.0`, but being supplied data created on `5.21` since the config field is set. Since we got so far as updating tables on the joiner, what's odd to me is that such a combination passed API/database version checking in LXD in the first place. 

I do remember from the initial Mattermost thread that the snaps were listed as `5.21`.

Worth a bit more investigation I think. @token47 are you able to reproduce this with any consistency?

Reviewer 3: I just gave this a try by running `microcloud init` on a system with lxd 5.21 installed, and lxd 5.0 installed on a joiner. Indeed this is the error I see:
```
Error: Failed to join cluster: Failed to initialize member: Failed to initialize storage pools and networks: Failed to update local member project "default": Invalid project configuration key "features.networks.zones"
```

So it seems we eagerly try to update the projects table before doing any API/database version checking in `5.0`, thus getting back a misleading/non-descriptive error. The real error is the daemon versions are incompatible.

Still not sure why `snap info` listed `5.21`, but maybe I just failed to catch a `5.0` in there.

Reviewer 2: Interestingly, I'm not being able to reproduce this anymore. Maybe something changed on LXD snap in the last days?

This is a test without upgrading lxd snap from 5.0 to 5.21 -> https://pastebin.canonical.com/p/mjpSWMhKmp/
This is a test with the refresh (not purge+install) -> https://pastebin.canonical.com/p/q75c9c25tc/

As you can see, second test was giving me error before but now seems to have worked.

Reviewer: @token47 IIRC the error in the first link seems to be caused by not having at least LXD 5.21.x.

If this still isn't reproducible anymore please let us know so we can close the PR and the related issue. LXD 5.21/stable channel was updated recently so maybe it contained a fix for this.

Reviewer 2: Sorry but have been unable to test this since. You seem to have hit the same issue, so maybe you could try one more time to make sure it was not a fluke for me? Maybe something actually changed that indirectly fixed this.

Reviewer: Just tested this again with three machines refreshing from `5.0/stable` to `5.21/stable`. It's working fine.

However, this is not recommended.
```

```{note}

> Depending on your machines OS and state, the LXD snap might already be installed (e.g you have a version of Ubuntu on the machine with LXD 5.0 already installed).

Reviewer 4: ```suggestion
It's possible that a required snap is already installed on your machine. For example, it might be running a version of Ubuntu that comes with LXD pre-installed.
```

Author: fixed

> In this case run `sudo snap refresh <snap> --channel=<track>/stable --cohort="+"`

Reviewer: Oops, just wanted the merge this, there is a dot missing at the end. Could you please add this one in?

> In this case run `sudo snap refresh <snap> --channel=<track>/stable --cohort="+"`

Reviewer 4: ```suggestion
In this case, run `sudo snap refresh <snap> --channel=<track>/stable --cohort="+"`
``` to refresh (update) the installed snap.

Author: fixed
